src/app/core/default-values/default-value.service.ts

```diff
@@ -109,6 +113,28 @@ export class DefaultValueService {
     } else {
       targetFormControl.setValue(fieldConfig.defaultValue.value);
     }
+    const staticDefaultValue = fieldConfig.defaultValue.value;
+    let transformedDefaultValue: any = staticDefaultValue;
+
+    if (fieldConfig.dataType === "configurable-enum") {
+      const enumValues = this.enumService.getEnumValues(fieldConfig.additional);
+      const matchedEnum = enumValues.find(
+        (enumValue) => enumValue.id === staticDefaultValue,
+      );
+
+      transformedDefaultValue = matchedEnum
+        ? this.entitySchemaService.valueToEntityFormat(
+            staticDefaultValue,
+            fieldConfig,
+          )
```

Comment:
have you checked what `this.entitySchemaService.valueToEntityFormat(staticDefaultValue, fieldConfig)` returns in the cases that you handle specially here (i.e. when no matchedEnum exists and you run valueToEntityFormat; and if fieldConfig.isArray is set)? My expectation would be that `entitySchemaService.valueToEntityFormat()` already returns sensible results for these automatically and the extra logic here is not needed.

I also think we can run this for any fieldConfig.dataType - there should not be issues for other defaultValues (and this may even prepare for other special defaults)

Reply:
1. I added the matchEnum check because, if matchEnum does not exist, the selected field value appears as [invalid option], which I felt was not a sensible result. That’s why I included this check. - <img width="357" alt="image" src="https://github.com/user-attachments/assets/eba77482-74a8-4710-8db9-4c820d7a79a3" />
2. Regarding fieldConfig.dataType, I agree with your point and will remove that.